pgx-examples/composite_types/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+use pgx::*;
+
+pg_module_magic!();
+
+#[allow(dead_code)]
+#[derive(Debug)]
+struct MyType {
+    name: Option<String>,
+    age: Option<i32>,
+}
+
+impl FromDatum for MyType {
+    unsafe fn from_datum(composite: pg_sys::Datum, is_null: bool) -> Option<Self>
+    where
+        Self: Sized,
+    {
+        if is_null {
+            None
+        } else {
+            let tuple = PgHeapTuple::from_composite_datum(composite);
+            Some(Self {
+                name: tuple
+                    .get_by_name::<String>("name")
+                    .expect("failed to get attribute: name"),
+                age: tuple
+                    .get_by_name::<i32>("age")
+                    .expect("failed to get attribute: age"),
+            })
+        }
+    }
+}
+
+#[pg_extern]
+unsafe fn debug_my_type(
+    _my_type: AnyElement,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<AnyElement> {
+    let composite = pg_getarg_datum(fcinfo, 0).unwrap();
+    let mut composite = PgHeapTuple::from_composite_datum(composite);
+
+    composite
+        .set_by_name("age", 100)
+        .expect("no such attribute");
+    composite
+        .set_by_name("name", "Someone Else")
+        .expect("no such attribute");
+
+    let composite = composite.into_composite_datum().unwrap();
+    AnyElement::from_datum(composite, false)
+}
+
+#[pg_extern]
+fn hello_composite_types() -> &'static str {
```

Comment:
Probably can be removed as it's very unrelated?

Reply:
I'm gonna yank this example from this PR.  This "composite type" stuff ain't quite finished yet and really should be part of the work I do for #499.